flexmeasures/auth/policy.py

```diff
@@ -226,13 +227,21 @@ def can_modify_role(user, roles_to_modify) -> bool:
             return False
         if role.name == ADMIN_READER_ROLE and not user.has_role(ADMIN_ROLE):
             return False
-        if role.name == ACCOUNT_ADMIN_ROLE and not (
-            user.has_role(ADMIN_ROLE) or user.has_role(CONSULTANT_ROLE)
-        ):
-            return False
-        if role.name == CONSULTANT_ROLE and not (
-            user.has_role(ADMIN_ROLE) or user.has_role(ACCOUNT_ADMIN_ROLE)
-        ):
-            return False
+        if role.name == ACCOUNT_ADMIN_ROLE:
+            if user.has_role(ADMIN_ROLE):
+                return True
+
+            if not user.has_role(CONSULTANT_ROLE) and (
+                user.account != modified_user.account.consultancy_account
```

Comment:
Good find with the object comparison problem. 

Then it seems a good idea to also compare IDs here, right?